.github/workflows/build_and_test.yml

```diff
@@ -1,39 +1,79 @@
 # This workflow will build a Java project with Maven
 # For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
 
-name: Java CI with Maven
+name: Build
 
 on:
   push:
 
+env: 
+  IS_NON_FEATURE_BRANCH: ${{ github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release') }}
+
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         # test against latest update of each major Java version, as well as specific updates of LTS versions:
         java: [8, 11.0.x, 13 ]
-
+    
     steps:
-    - uses: actions/checkout@v2
-      
-    - uses: actions/cache@v1
+    - name: Checkout the code
+      uses: actions/checkout@v2
+
+    - name: Cache the maven artifacts
+      uses: actions/cache@v1
       with:
         path: ~/.m2/repository
         key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
         restore-keys: |
           ${{ runner.os }}-maven-
           
-    - uses: actions/setup-java@v1
+    - name: Configure java
+      uses: actions/setup-java@v1
       with:
         java-version: ${{ matrix.java }}
 
-    - name: Build with Maven
-      run: mvn -B package --file pom.xml
+    - name: Build and run tests
+      run: mvn -B package 
+
+    - run: echo ${{ env.IS_NON_FEATURE_BRANCH }}
+
+    - name: Run integration tests
+      run: mvn verify
+      if: env.IS_NON_FEATURE_BRANCH == 'true'
+      env:
+        REVAI_ACCESS_TOKEN: ${{ secrets.INTEGRATION_TEST_ACCESS_TOKEN }}
     
-    - run: mkdir staging && cp target/*.jar staging
+    - name: Save build artifacts to staging area
+      run: mkdir staging && cp target/*.jar staging
     
-    - uses: actions/upload-artifact@v1
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v1
       with:
         name: Package
         path: staging
+    
+    - name: Notify slack fail
```

Comment:
Can we include the version that failed in this notification?

Reply:
yeah it does already, you can see how it looks in the channel - https://rev.slack.com/archives/CFPT8VA2Z/p1590010926249400. it includes the branch name and the link includes the commit